src/webgpu/shader/validation/types/vector.spec.ts

```diff
@@ -0,0 +1,68 @@
+export const description = `
+Validation tests for vector types
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kCases = {
+  // Valid vector types
+  vec2_i32: { wgsl: 'alias T = vec2<i32>;', ok: true },
+  vec3_i32: { wgsl: 'alias T = vec3<i32>;', ok: true },
+  vec4_i32: { wgsl: 'alias T = vec4<i32>;', ok: true },
+  vec2_u32: { wgsl: 'alias T = vec2<u32>;', ok: true },
+  vec3_u32: { wgsl: 'alias T = vec3<u32>;', ok: true },
+  vec4_u32: { wgsl: 'alias T = vec4<u32>;', ok: true },
+  vec2_f32: { wgsl: 'alias T = vec2<f32>;', ok: true },
+  vec3_f32: { wgsl: 'alias T = vec3<f32>;', ok: true },
+  vec4_f32: { wgsl: 'alias T = vec4<f32>;', ok: true },
+  vec2_f16: { wgsl: 'enable f16;\nalias T = vec2<f16>;', ok: true },
+  vec3_f16: { wgsl: 'enable f16;\nalias T = vec3<f16>;', ok: true },
+  vec4_f16: { wgsl: 'enable f16;\nalias T = vec4<f16>;', ok: true },
+
+  // Pre-declared type aliases
+  vec2i: { wgsl: 'const c : vec2i = vec2<i32>();', ok: true },
+  vec3i: { wgsl: 'const c : vec3i = vec3<i32>();', ok: true },
+  vec4i: { wgsl: 'const c : vec4i = vec4<i32>();', ok: true },
+  vec2u: { wgsl: 'const c : vec2u = vec2<u32>();', ok: true },
+  vec3u: { wgsl: 'const c : vec3u = vec3<u32>();', ok: true },
+  vec4u: { wgsl: 'const c : vec4u = vec4<u32>();', ok: true },
+  vec2f: { wgsl: 'const c : vec2f = vec2<f32>();', ok: true },
+  vec3f: { wgsl: 'const c : vec3f = vec3<f32>();', ok: true },
+  vec4f: { wgsl: 'const c : vec4f = vec4<f32>();', ok: true },
+  vec2h: { wgsl: 'enable f16;\nconst c : vec2h = vec2<f16>();', ok: true },
+  vec3h: { wgsl: 'enable f16;\nconst c : vec3h = vec3<f16>();', ok: true },
+  vec4h: { wgsl: 'enable f16;\nconst c : vec4h = vec4<f16>();', ok: true },
+
+  // pass
+  trailing_comma: { wgsl: 'alias T = vec3<u32,>;', ok: true },
+  aliased_el_ty: { wgsl: 'alias EL = i32;\nalias T = vec3<EL>;', ok: true },
+
+  // invalid
+  vec: { wgsl: 'alias T = vec;', ok: false },
```

Comment:
Please test trying to make vector types where the element type is:
- an array (sized, unsized)
- structure
- an atomic<u32>
- a matrix
- a vector type.

Reply:
Done